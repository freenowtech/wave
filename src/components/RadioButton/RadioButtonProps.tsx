import { ComponentPropsWithoutRef, ReactNode } from 'react';
import { MarginProps } from 'styled-system';
import { ClassNameProps } from '../../utils/extractProps';

interface RadioButtonProps extends ComponentPropsWithoutRef<'input'>, ClassNameProps, MarginProps {
    /**
     * Provide a label for the input which will be shown next to the radio button
     */
    label?: ReactNode;
    /**
     * Add styles to show the radio button has errors
     */
    error?: boolean;
    /**
     * Align the label text relatively to the radio button
     */
    textVerticalAlign?: 'top' | 'center';
}

export { RadioButtonProps };
