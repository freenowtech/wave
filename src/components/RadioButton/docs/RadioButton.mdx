---
name: RadioButton
menu: Components
route: /components/RadioButton
---

import { Playground } from 'docz';
import { RadioButton } from '../RadioButton';
import { RadioButtonPropsTable } from './RadioButtonPropsTable';
import { Combination } from '../../../docs/Combination'
import { StyledSystemLinks } from '../../../docs/StyledSystemLinks'

# RadioButton

Use radio buttons when you have a few options that a user can choose from. Never use radio buttons if the user can
select more than one option from a list (consider using a [Checkbox](/components/Checkbox) instead).

You can provide the label as either a `string` or as a react component. If a `string` is provided as the label, it
will be wrapped with the [Text](/components/text) component.


## Properties

<RadioButtonPropsTable />
<br />

<StyledSystemLinks component="RadioButton" supportedProps={['margin']} />

## Examples

<Playground>
    <RadioButton name="example" label="Taxi" />
    <br />
    <RadioButton name="example" label="Scooter" />
    <br />
    <RadioButton name="example" label="Bike" />
</Playground>

## Combinations

<Combination error={[false, true]} disabled={[false, true]}>
    {(props, i) => <RadioButton key={i} {...props} label="Option" />}
</Combination>
