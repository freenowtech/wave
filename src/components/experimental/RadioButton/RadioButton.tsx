import React, { FC } from 'react';
import styled from 'styled-components';
import { Radio as BaseRadio, RadioProps } from 'react-aria-components';

import { getSemanticValue } from '../../../essentials/experimental';
import { themeGet } from '../../../utils/experimental';

const Indicator = styled.span<{ $selected: boolean }>`
    position: relative;
    flex-shrink: 0;

    top: calc((var(--wave-exp-typescale-body-1-line-height) - 1rem) / 2);
    width: 1rem;
    height: 1rem;

    box-sizing: border-box;
    border-style: solid;
    border-color: currentColor;
    border-width: ${props => (props.$selected ? '5px' : '2px')};
    border-radius: 50%;
    background-color: ${getSemanticValue('surface')};
    transition: border-color 200ms ease, border-width 50ms ease;
`;

const Radio = styled(BaseRadio)`
    display: flex;
    gap: ${themeGet('space.2')};
    cursor: pointer;

    font-family: inherit;
    font-size: var(--wave-exp-typescale-body-1-size);
    font-weight: var(--wave-exp-typescale-body-1-weight);
    line-height: var(--wave-exp-typescale-body-1-line-height);

    color: ${getSemanticValue('on-surface')};

    ${Indicator} {
        color: ${getSemanticValue('divider')};
    }

    &[data-hovered] ${Indicator} {
        color: ${getSemanticValue('interactive')};
    }

    &[data-pressed] ${Indicator} {
        color: ${getSemanticValue('surface-variant')};
    }

    &[data-focus-visible] {
        outline: 2px solid ${getSemanticValue('surface-variant')};
        outline-offset: 2px;
    }

    &[data-disabled] {
        cursor: not-allowed;
        opacity: 0.38;
    }

    &[data-invalid] {
        color: ${getSemanticValue('negative-variant')};
    }

    &[data-hovered][data-invalid] ${Indicator} {
        color: ${getSemanticValue('negative')};
    }

    &[data-selected] ${Indicator} {
        color: ${getSemanticValue('accent')};
    }

    &[data-selected][data-hovered] ${Indicator} {
        color: ${getSemanticValue('on-interactive-container')};
    }

    &[data-selected][data-pressed] ${Indicator} {
        color: ${getSemanticValue('interactive')};
    }

    &[data-selected][data-disabled] ${Indicator} {
        color: ${getSemanticValue('surface-variant')};
    }
`;

export const RadioButton: FC<RadioProps> = ({ children, ...rest }) => (
    <Radio {...rest}>
        {({ isSelected }) => (
            <>
                <Indicator $selected={isSelected} />
                {children}
            </>
        )}
    </Radio>
);
