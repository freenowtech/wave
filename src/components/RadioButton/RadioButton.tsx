import React, { FC, ReactNode } from 'react';
import styled from 'styled-components';

import { extractClassNameProps, extractWrapperMarginProps } from '../../utils/extractProps';
import { Text } from '../Text/Text';

import { Checkmark } from './components/Checkmark';
import { LabelWrapper } from './components/LabelWrapper';
import { TapArea } from './components/TapArea';
import { RadioButtonProps } from './RadioButtonProps';

const WithTapAreaWrapper = styled.div<Pick<RadioButtonProps, 'textVerticalAlign'>>`
    width: 1.5rem;
    position: relative;
    display: inline-flex;
    align-items: center;

    ${({ textVerticalAlign }) => (textVerticalAlign === 'top' ? 'margin-top: 0.1875rem' : undefined)}
`;

const RadioButton: FC<RadioButtonProps> = ({ textVerticalAlign = 'center', ...props }: RadioButtonProps) => {
    const { classNameProps, restProps: withoutClassName } = extractClassNameProps(props);
    const { marginProps, restProps } = extractWrapperMarginProps(withoutClassName);
    const { disabled, error, label, ...rest } = restProps;

    let dynamicLabel: ReactNode = label;

    if (typeof label === 'string') {
        dynamicLabel = <Text>{label}</Text>;
    }

    return (
        <LabelWrapper
            disabled={disabled}
            error={error}
            textVerticalAlign={textVerticalAlign}
            {...marginProps}
            {...classNameProps}
        >
            <WithTapAreaWrapper textVerticalAlign={textVerticalAlign}>
                <TapArea />
                <Checkmark type="radio" disabled={disabled} error={error} {...rest} />
            </WithTapAreaWrapper>
            {dynamicLabel}
        </LabelWrapper>
    );
};

export { RadioButton };
