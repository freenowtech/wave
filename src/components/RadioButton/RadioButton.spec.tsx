import { render } from '@testing-library/react';
import * as React from 'react';
import { RadioButton } from './RadioButton';

describe('Checkbox', () => {
    it('renders with default props', () => {
        expect(render(<RadioButton />).container.firstChild).toMatchSnapshot();
    });

    it('renders the checked style', () => {
        expect(render(<RadioButton defaultChecked />).container.firstChild).toMatchSnapshot();
    });

    it('renders the error style', () => {
        expect(render(<RadioButton error />).container.firstChild).toMatchSnapshot();
    });

    it('renders the error checked style', () => {
        expect(render(<RadioButton error disabled />).container.firstChild).toMatchSnapshot();
    });

    it('renders the disabled style', () => {
        expect(render(<RadioButton disabled />).container.firstChild).toMatchSnapshot();
    });

    it('renders the disabled checked style', () => {
        expect(render(<RadioButton disabled defaultChecked />).container.firstChild).toMatchSnapshot();
    });

    it('renders the correct text vertical align version', () => {
        expect(render(<RadioButton label="Test" textVerticalAlign="top" />).container.firstChild).toMatchSnapshot();
    });
});
