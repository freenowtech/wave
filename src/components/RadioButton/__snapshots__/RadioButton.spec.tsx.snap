// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox renders the checked style 1`] = `
.c5 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  outline: none;
  border: 0;
  position: relative;
  width: 1rem;
  height: 1rem;
  padding: 0;
  margin: 0 0.5rem 0 0;
  background-color: #FFFFFF;
  box-shadow: inset 0 0 0 0.125rem #C6CDD4;
  border-radius: 50%;
  -webkit-transition: background-color 100ms,box-shadow 100ms;
  transition: background-color 100ms,box-shadow 100ms;
  cursor: pointer;
  vertical-align: text-bottom;
}

.c5::after {
  content: ' ';
  width: 1rem;
  height: 1rem;
  position: absolute;
  opacity: 0;
  visibility: hidden;
  -webkit-transform: scale(0.2);
  -ms-transform: scale(0.2);
  transform: scale(0.2);
  -webkit-transition: visibility 175ms,opacity 150ms,scale 175ms;
  transition: visibility 175ms,opacity 150ms,scale 175ms;
}

.c5:checked {
  box-shadow: inset 0 0 0 0.3125rem #096BDB;
}

.c5:checked::after {
  opacity: 1;
  visibility: visible;
  -webkit-transform: scale(1);
  -ms-transform: scale(1);
  transform: scale(1);
}

.c5:disabled {
  cursor: not-allowed;
  background-color: #F1F2F4;
  box-shadow: inset 0 0 0 0.125rem #F1F2F4;
}

.c5:disabled:hover {
  box-shadow: inset 0 0 0 0.125rem #F1F2F4;
}

.c5:disabled:active {
  background-color: #F1F2F4;
}

.c5:disabled:checked {
  box-shadow: inset 0 0 0 0.3125rem #F1F2F4;
  background-color: #FFFFFF;
}

.c3 {
  position: absolute;
  width: 1.75rem;
  height: 1.75rem;
  border-radius: 50%;
  background-color: transparent;
  top: -0.375rem;
  left: -0.375rem;
  -webkit-transition: background-color 125ms;
  transition: background-color 125ms;
}

.c3:active {
  background-color: #F1F2F4;
}

.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  color: #001E3E;
  font-family: "Open Sans",sans-serif;
  line-height: 1;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0:hover {
  cursor: pointer;
}

.c0:hover .c2:not(:active) {
  background-color: #F1F7FD;
}

.c0:hover .c4:not(:checked) {
  box-shadow: inset 0 0 0 0.125rem #096BDB;
  background-color: #FFFFFF;
}

.c1 {
  width: 1.5rem;
  position: relative;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

<label
  class="c0"
>
  <div
    class="c1"
  >
    <div
      class="c2 c3"
    />
    <input
      checked=""
      class="c4 c5"
      type="radio"
    />
  </div>
</label>
`;

exports[`Checkbox renders the correct text vertical align version 1`] = `
.c6 {
  color: #001E3E;
  font-size: 1rem;
  font-family: "Open Sans",sans-serif;
  line-height: 1.4;
  margin: 0;
}

.c5 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  outline: none;
  border: 0;
  position: relative;
  width: 1rem;
  height: 1rem;
  padding: 0;
  margin: 0 0.5rem 0 0;
  background-color: #FFFFFF;
  box-shadow: inset 0 0 0 0.125rem #C6CDD4;
  border-radius: 50%;
  -webkit-transition: background-color 100ms,box-shadow 100ms;
  transition: background-color 100ms,box-shadow 100ms;
  cursor: pointer;
  vertical-align: text-bottom;
}

.c5::after {
  content: ' ';
  width: 1rem;
  height: 1rem;
  position: absolute;
  opacity: 0;
  visibility: hidden;
  -webkit-transform: scale(0.2);
  -ms-transform: scale(0.2);
  transform: scale(0.2);
  -webkit-transition: visibility 175ms,opacity 150ms,scale 175ms;
  transition: visibility 175ms,opacity 150ms,scale 175ms;
}

.c5:checked {
  box-shadow: inset 0 0 0 0.3125rem #096BDB;
}

.c5:checked::after {
  opacity: 1;
  visibility: visible;
  -webkit-transform: scale(1);
  -ms-transform: scale(1);
  transform: scale(1);
}

.c5:disabled {
  cursor: not-allowed;
  background-color: #F1F2F4;
  box-shadow: inset 0 0 0 0.125rem #F1F2F4;
}

.c5:disabled:hover {
  box-shadow: inset 0 0 0 0.125rem #F1F2F4;
}

.c5:disabled:active {
  background-color: #F1F2F4;
}

.c5:disabled:checked {
  box-shadow: inset 0 0 0 0.3125rem #F1F2F4;
  background-color: #FFFFFF;
}

.c3 {
  position: absolute;
  width: 1.75rem;
  height: 1.75rem;
  border-radius: 50%;
  background-color: transparent;
  top: -0.375rem;
  left: -0.375rem;
  -webkit-transition: background-color 125ms;
  transition: background-color 125ms;
}

.c3:active {
  background-color: #F1F2F4;
}

.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  color: #001E3E;
  font-family: "Open Sans",sans-serif;
  line-height: 1;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

.c0:hover {
  cursor: pointer;
}

.c0:hover .c2:not(:active) {
  background-color: #F1F7FD;
}

.c0:hover .c4:not(:checked) {
  box-shadow: inset 0 0 0 0.125rem #096BDB;
  background-color: #FFFFFF;
}

.c1 {
  width: 1.5rem;
  position: relative;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-top: 0.1875rem;
}

<label
  class="c0"
>
  <div
    class="c1"
  >
    <div
      class="c2 c3"
    />
    <input
      class="c4 c5"
      type="radio"
    />
  </div>
  <span
    class="c6"
  >
    Test
  </span>
</label>
`;

exports[`Checkbox renders the disabled checked style 1`] = `
.c4 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  outline: none;
  border: 0;
  position: relative;
  width: 1rem;
  height: 1rem;
  padding: 0;
  margin: 0 0.5rem 0 0;
  background-color: #FFFFFF;
  box-shadow: inset 0 0 0 0.125rem #C6CDD4;
  border-radius: 50%;
  -webkit-transition: background-color 100ms,box-shadow 100ms;
  transition: background-color 100ms,box-shadow 100ms;
  cursor: pointer;
  vertical-align: text-bottom;
}

.c4::after {
  content: ' ';
  width: 1rem;
  height: 1rem;
  position: absolute;
  opacity: 0;
  visibility: hidden;
  -webkit-transform: scale(0.2);
  -ms-transform: scale(0.2);
  transform: scale(0.2);
  -webkit-transition: visibility 175ms,opacity 150ms,scale 175ms;
  transition: visibility 175ms,opacity 150ms,scale 175ms;
}

.c4:checked {
  box-shadow: inset 0 0 0 0.3125rem #096BDB;
}

.c4:checked::after {
  opacity: 1;
  visibility: visible;
  -webkit-transform: scale(1);
  -ms-transform: scale(1);
  transform: scale(1);
}

.c4:disabled {
  cursor: not-allowed;
  background-color: #F1F2F4;
  box-shadow: inset 0 0 0 0.125rem #F1F2F4;
}

.c4:disabled:hover {
  box-shadow: inset 0 0 0 0.125rem #F1F2F4;
}

.c4:disabled:active {
  background-color: #F1F2F4;
}

.c4:disabled:checked {
  box-shadow: inset 0 0 0 0.3125rem #F1F2F4;
  background-color: #FFFFFF;
}

.c3 {
  position: absolute;
  width: 1.75rem;
  height: 1.75rem;
  border-radius: 50%;
  background-color: transparent;
  top: -0.375rem;
  left: -0.375rem;
  -webkit-transition: background-color 125ms;
  transition: background-color 125ms;
}

.c3:active {
  background-color: #F1F2F4;
}

.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  color: #9CA7B4;
  font-family: "Open Sans",sans-serif;
  line-height: 1;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0:hover {
  cursor: not-allowed;
}

.c0:hover .c2:active {
  background-color: transparent;
}

.c1 {
  width: 1.5rem;
  position: relative;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

<label
  class="c0"
  disabled=""
>
  <div
    class="c1"
  >
    <div
      class="c2 c3"
    />
    <input
      checked=""
      class="c4"
      disabled=""
      type="radio"
    />
  </div>
</label>
`;

exports[`Checkbox renders the disabled style 1`] = `
.c4 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  outline: none;
  border: 0;
  position: relative;
  width: 1rem;
  height: 1rem;
  padding: 0;
  margin: 0 0.5rem 0 0;
  background-color: #FFFFFF;
  box-shadow: inset 0 0 0 0.125rem #C6CDD4;
  border-radius: 50%;
  -webkit-transition: background-color 100ms,box-shadow 100ms;
  transition: background-color 100ms,box-shadow 100ms;
  cursor: pointer;
  vertical-align: text-bottom;
}

.c4::after {
  content: ' ';
  width: 1rem;
  height: 1rem;
  position: absolute;
  opacity: 0;
  visibility: hidden;
  -webkit-transform: scale(0.2);
  -ms-transform: scale(0.2);
  transform: scale(0.2);
  -webkit-transition: visibility 175ms,opacity 150ms,scale 175ms;
  transition: visibility 175ms,opacity 150ms,scale 175ms;
}

.c4:checked {
  box-shadow: inset 0 0 0 0.3125rem #096BDB;
}

.c4:checked::after {
  opacity: 1;
  visibility: visible;
  -webkit-transform: scale(1);
  -ms-transform: scale(1);
  transform: scale(1);
}

.c4:disabled {
  cursor: not-allowed;
  background-color: #F1F2F4;
  box-shadow: inset 0 0 0 0.125rem #F1F2F4;
}

.c4:disabled:hover {
  box-shadow: inset 0 0 0 0.125rem #F1F2F4;
}

.c4:disabled:active {
  background-color: #F1F2F4;
}

.c4:disabled:checked {
  box-shadow: inset 0 0 0 0.3125rem #F1F2F4;
  background-color: #FFFFFF;
}

.c3 {
  position: absolute;
  width: 1.75rem;
  height: 1.75rem;
  border-radius: 50%;
  background-color: transparent;
  top: -0.375rem;
  left: -0.375rem;
  -webkit-transition: background-color 125ms;
  transition: background-color 125ms;
}

.c3:active {
  background-color: #F1F2F4;
}

.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  color: #9CA7B4;
  font-family: "Open Sans",sans-serif;
  line-height: 1;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0:hover {
  cursor: not-allowed;
}

.c0:hover .c2:active {
  background-color: transparent;
}

.c1 {
  width: 1.5rem;
  position: relative;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

<label
  class="c0"
  disabled=""
>
  <div
    class="c1"
  >
    <div
      class="c2 c3"
    />
    <input
      class="c4"
      disabled=""
      type="radio"
    />
  </div>
</label>
`;

exports[`Checkbox renders the error checked style 1`] = `
.c4 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  outline: none;
  border: 0;
  position: relative;
  width: 1rem;
  height: 1rem;
  padding: 0;
  margin: 0 0.5rem 0 0;
  background-color: #FFFFFF;
  box-shadow: inset 0 0 0 0.125rem #FF5E05;
  border-radius: 50%;
  -webkit-transition: background-color 100ms,box-shadow 100ms;
  transition: background-color 100ms,box-shadow 100ms;
  cursor: pointer;
  vertical-align: text-bottom;
}

.c4::after {
  content: ' ';
  width: 1rem;
  height: 1rem;
  position: absolute;
  opacity: 0;
  visibility: hidden;
  -webkit-transform: scale(0.2);
  -ms-transform: scale(0.2);
  transform: scale(0.2);
  -webkit-transition: visibility 175ms,opacity 150ms,scale 175ms;
  transition: visibility 175ms,opacity 150ms,scale 175ms;
}

.c4:checked {
  box-shadow: inset 0 0 0 0.3125rem #FF5E05;
}

.c4:checked::after {
  opacity: 1;
  visibility: visible;
  -webkit-transform: scale(1);
  -ms-transform: scale(1);
  transform: scale(1);
}

.c4:disabled {
  cursor: not-allowed;
  background-color: #F1F2F4;
  box-shadow: inset 0 0 0 0.125rem #F1F2F4;
}

.c4:disabled:hover {
  box-shadow: inset 0 0 0 0.125rem #F1F2F4;
}

.c4:disabled:active {
  background-color: #F1F2F4;
}

.c4:disabled:checked {
  box-shadow: inset 0 0 0 0.3125rem #F1F2F4;
  background-color: #FFFFFF;
}

.c3 {
  position: absolute;
  width: 1.75rem;
  height: 1.75rem;
  border-radius: 50%;
  background-color: transparent;
  top: -0.375rem;
  left: -0.375rem;
  -webkit-transition: background-color 125ms;
  transition: background-color 125ms;
}

.c3:active {
  background-color: #F1F2F4;
}

.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  color: #9CA7B4;
  font-family: "Open Sans",sans-serif;
  line-height: 1;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0:hover {
  cursor: not-allowed;
}

.c0:hover .c2:active {
  background-color: transparent;
}

.c1 {
  width: 1.5rem;
  position: relative;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

<label
  class="c0"
  disabled=""
>
  <div
    class="c1"
  >
    <div
      class="c2 c3"
    />
    <input
      class="c4"
      disabled=""
      type="radio"
    />
  </div>
</label>
`;

exports[`Checkbox renders the error style 1`] = `
.c5 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  outline: none;
  border: 0;
  position: relative;
  width: 1rem;
  height: 1rem;
  padding: 0;
  margin: 0 0.5rem 0 0;
  background-color: #FFFFFF;
  box-shadow: inset 0 0 0 0.125rem #FF5E05;
  border-radius: 50%;
  -webkit-transition: background-color 100ms,box-shadow 100ms;
  transition: background-color 100ms,box-shadow 100ms;
  cursor: pointer;
  vertical-align: text-bottom;
}

.c5::after {
  content: ' ';
  width: 1rem;
  height: 1rem;
  position: absolute;
  opacity: 0;
  visibility: hidden;
  -webkit-transform: scale(0.2);
  -ms-transform: scale(0.2);
  transform: scale(0.2);
  -webkit-transition: visibility 175ms,opacity 150ms,scale 175ms;
  transition: visibility 175ms,opacity 150ms,scale 175ms;
}

.c5:checked {
  box-shadow: inset 0 0 0 0.3125rem #FF5E05;
}

.c5:checked::after {
  opacity: 1;
  visibility: visible;
  -webkit-transform: scale(1);
  -ms-transform: scale(1);
  transform: scale(1);
}

.c5:disabled {
  cursor: not-allowed;
  background-color: #F1F2F4;
  box-shadow: inset 0 0 0 0.125rem #F1F2F4;
}

.c5:disabled:hover {
  box-shadow: inset 0 0 0 0.125rem #F1F2F4;
}

.c5:disabled:active {
  background-color: #F1F2F4;
}

.c5:disabled:checked {
  box-shadow: inset 0 0 0 0.3125rem #F1F2F4;
  background-color: #FFFFFF;
}

.c3 {
  position: absolute;
  width: 1.75rem;
  height: 1.75rem;
  border-radius: 50%;
  background-color: transparent;
  top: -0.375rem;
  left: -0.375rem;
  -webkit-transition: background-color 125ms;
  transition: background-color 125ms;
}

.c3:active {
  background-color: #F1F2F4;
}

.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  color: #001E3E;
  font-family: "Open Sans",sans-serif;
  line-height: 1;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0:hover {
  cursor: pointer;
}

.c0:hover .c2:not(:active) {
  background-color: #FFF6F1;
}

.c0:hover .c4:not(:checked) {
  box-shadow: inset 0 0 0 0.125rem #FF5E05;
  background-color: #FFFFFF;
}

.c1 {
  width: 1.5rem;
  position: relative;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

<label
  class="c0"
>
  <div
    class="c1"
  >
    <div
      class="c2 c3"
    />
    <input
      class="c4 c5"
      type="radio"
    />
  </div>
</label>
`;

exports[`Checkbox renders with default props 1`] = `
.c5 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  outline: none;
  border: 0;
  position: relative;
  width: 1rem;
  height: 1rem;
  padding: 0;
  margin: 0 0.5rem 0 0;
  background-color: #FFFFFF;
  box-shadow: inset 0 0 0 0.125rem #C6CDD4;
  border-radius: 50%;
  -webkit-transition: background-color 100ms,box-shadow 100ms;
  transition: background-color 100ms,box-shadow 100ms;
  cursor: pointer;
  vertical-align: text-bottom;
}

.c5::after {
  content: ' ';
  width: 1rem;
  height: 1rem;
  position: absolute;
  opacity: 0;
  visibility: hidden;
  -webkit-transform: scale(0.2);
  -ms-transform: scale(0.2);
  transform: scale(0.2);
  -webkit-transition: visibility 175ms,opacity 150ms,scale 175ms;
  transition: visibility 175ms,opacity 150ms,scale 175ms;
}

.c5:checked {
  box-shadow: inset 0 0 0 0.3125rem #096BDB;
}

.c5:checked::after {
  opacity: 1;
  visibility: visible;
  -webkit-transform: scale(1);
  -ms-transform: scale(1);
  transform: scale(1);
}

.c5:disabled {
  cursor: not-allowed;
  background-color: #F1F2F4;
  box-shadow: inset 0 0 0 0.125rem #F1F2F4;
}

.c5:disabled:hover {
  box-shadow: inset 0 0 0 0.125rem #F1F2F4;
}

.c5:disabled:active {
  background-color: #F1F2F4;
}

.c5:disabled:checked {
  box-shadow: inset 0 0 0 0.3125rem #F1F2F4;
  background-color: #FFFFFF;
}

.c3 {
  position: absolute;
  width: 1.75rem;
  height: 1.75rem;
  border-radius: 50%;
  background-color: transparent;
  top: -0.375rem;
  left: -0.375rem;
  -webkit-transition: background-color 125ms;
  transition: background-color 125ms;
}

.c3:active {
  background-color: #F1F2F4;
}

.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  color: #001E3E;
  font-family: "Open Sans",sans-serif;
  line-height: 1;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0:hover {
  cursor: pointer;
}

.c0:hover .c2:not(:active) {
  background-color: #F1F7FD;
}

.c0:hover .c4:not(:checked) {
  box-shadow: inset 0 0 0 0.125rem #096BDB;
  background-color: #FFFFFF;
}

.c1 {
  width: 1.5rem;
  position: relative;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

<label
  class="c0"
>
  <div
    class="c1"
  >
    <div
      class="c2 c3"
    />
    <input
      class="c4 c5"
      type="radio"
    />
  </div>
</label>
`;
